modules/database/src/ioc/db/recGbl.c

```diff
@@ -44,8 +44,11 @@
 #include "dbScan.h"
 #include "devSup.h"
 #include "link.h"
-#include "recGbl.h"
 
+/* avoid warning about NULL format argument to recGblSetSevrMsg */
+#undef EPICS_PRINTF_STYLE
+#define EPICS_PRINTF_STYLE(a,b)
+#include "recGbl.h"
```

Comment:
I don't quite understand how this helps. It applies to this file, but there are no calls to `recGblSetSevrMsg` in here. The warning would still be generated by the call sites in other files, no? Or did I misunderstand the cause of the warning?

Reply:
Ah, found it, my bad. Rather than `undef`ing the macro, I suggest replacing `NULL` with an empty string.

Reply:
Originally, gcc 2.96 complained:
```
../db/recGbl.c:256: warning: null format string
```
Passing an empty string warns on several other compilers (gcc 4.2.2 and higher):
```
../db/recGbl.c:256:5: warning: zero-length gnu_printf format string [-Wformat-zero-length]
     return recGblSetSevrMsg(precord, new_stat, new_sevr, "");
```